packages/generator/src/action-function-import/action.ts

```diff
@@ -7,15 +7,29 @@ import { additionalDocForEntityNotDeserializable } from './function';
 
 const parameterName = 'parameters';
 
+function actionImportReturnType(actionImport: VdmActionImport) {
+  let type = actionImport.returnType.returnType;
+
+  if (actionImport.returnType.isCollection) {
+    type = `${type}[]`;
```

Comment:
question: when it's collection, it's implicitly non-nullable?

Reply:
same question to function as well

Reply:
no it will add the `|null` a few lines later.